### ISSUE TITLE
android: Unable to find method 'com.android.build.gradle.api.BaseVariant.getJavaCompiler()

### ISSUE THREAD

**Issue by tmtron** [xtext/xtext-gradle-plugin#73]
I'm just getting started with xtext and I've written a test-DSL in xtext and generated an IntelliJ plugin. I have a small Android-test project and in Android Studio my DSL-editor shows up and also generates the output files as expected.  
Now I try to configure gradle to also generate the files, but this fails with an error.

I found the Android Integration section for the [xtext-builder](http://xtext.github.io/xtext-gradle-plugin/xtext-builder.html). This seems out-dated (e.g. it refers to org.xtext.android, which does not exist - also the link there is broken).

Anyway, in the [xtext-gradle-plugin github repo](https://github.com/xtext/xtext-gradle-plugin) there is an `org.xtext.android.builder` plugin: I guess this is the correct one.

My project build.gradle file looks like this (relevant parts only):

```gradle
buildscript {
    repositories {
        mavenLocal() // This is required so that the mydsl language is found
        maven {
            url "https://plugins.gradle.org/m2/" // needed for org.xtext:xtext-android-gradle-plugin
        }
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.2.3'
        classpath "org.xtext:xtext-android-gradle-plugin:1.0.15"
    }
}
```
when I activate the `org.xtext.android.builder` plugin in the `build.gradle` file of in my app module:
```gradle
apply plugin: 'com.android.application'
apply plugin: "org.xtext.android.builder" // causes the Error!
```
I get this error:

`Error:Unable to find method 'com.android.build.gradle.api.BaseVariant.getJavaCompiler()Lorg/gradle/api/tasks/compile/AbstractCompile;'...`
I guess there's some version mismatch or something is not up-to-date.

**Comment by oehme** [comment]
Yep, the return type of that method was changed in a binary incompatible way. We need to adjust the Xtext plugin to deal with the new return type.

Thanks for pointing me at the documentation problems!

**Comment by programadorthi** [comment]
@oehme , I have cloned the repository and I already solved this problem. I changed android gradle plugin version to 2.2.3 that is the recent version.

Updates in the xtext-android-gradle-plugin project.

build.gradle:
`compile 'com.android.tools.build:gradle:2.2.3'`

XtextAndroidBuilderPlugin.xtend:
From
`89 generatorTask.classpath = variant.javaCompiler.classpath.plus(project.files(android.bootClasspath))`
`90 generatorTask.classesDir = variant.javaCompiler.destinationDir`
To
`import org.gradle.api.tasks.compile.AbstractCompile`
`90 val compile = variant.javaCompiler as AbstractCompile`
`91 generatorTask.classpath = compile.classpath.plus(project.files(android.bootClasspath))`
`92 generatorTask.classesDir = compile.destinationDir`

The javaCompiler method returns a Task that does not have a classpath method.
I made the typecast to AbstractCompile and all worked
